include/SFML/Window/Keyboard.hpp

```diff
@@ -153,6 +158,158 @@ class SFML_WINDOW_API Keyboard
         KeyCount      ///< Keep last -- the total number of keyboard keys
     };
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Scancodes
+    ///
+    /// The enumerators are bound to a physical key and do *not* depend
+    /// on the keyboard layout used by the operating system. Usually, the AT-101
+    /// keyboard can be used as reference for the physical position of the keys.
+    ///
+    /// The scancodes are based on a subset of Table 12: Keyboard/Keypad Page
+    /// of Universal Serial Bus (USB): HID Usage Tables, v1.12.
+    ///
+    /// \todo When porting this for SFML 3, remove the `s` prefix and use
+    ///       enum class.
+    ///
+    ////////////////////////////////////////////////////////////
+    enum Scancode
+    {
+        ScanUnknown = -1,      ///< Represents any scancode not present in this enum
+        ScanA = 0,             ///< Keyboard a and A key
+        ScanB,                 ///< Keyboard b and B key
+        ScanC,                 ///< Keyboard c and C key
+        ScanD,                 ///< Keyboard d and D key
+        ScanE,                 ///< Keyboard e and E key
+        ScanF,                 ///< Keyboard f and F key
+        ScanG,                 ///< Keyboard g and G key
+        ScanH,                 ///< Keyboard h and H key
+        ScanI,                 ///< Keyboard i and I key
+        ScanJ,                 ///< Keyboard j and J key
+        ScanK,                 ///< Keyboard k and K key
+        ScanL,                 ///< Keyboard l and L key
+        ScanM,                 ///< Keyboard m and M key
+        ScanN,                 ///< Keyboard n and N key
+        ScanO,                 ///< Keyboard o and O key
+        ScanP,                 ///< Keyboard p and P key
+        ScanQ,                 ///< Keyboard q and Q key
+        ScanR,                 ///< Keyboard r and R key
+        ScanS,                 ///< Keyboard s and S key
+        ScanT,                 ///< Keyboard t and T key
+        ScanU,                 ///< Keyboard u and U key
+        ScanV,                 ///< Keyboard v and V key
+        ScanW,                 ///< Keyboard w and W key
+        ScanX,                 ///< Keyboard x and X key
+        ScanY,                 ///< Keyboard y and Y key
+        ScanZ,                 ///< Keyboard z and Z key
+        ScanNum1,              ///< Keyboard 1 and ! key
```

Comment:
Do we really need to say "X and Y" for scancodes? On some layouts this is "1 and |". On some it's "1 and &". Maybe we need to describe scancodes by their main function.

Reply:
The scan key represents the key and not what it can type so upper and lower case makes no sense here, I agree.

Reply:
The doc corresponds to the AT-101 reference keyboard. It can give the user an idea of the physical location of each key.

Reply:
Yes, it's in reference to the AT-101 reference, we may want to think about potentially adding an image or similar, that helps better understand the layout and mapping